starknet_py/tests/e2e/account/account_test.py

```diff
@@ -178,3 +187,138 @@ async def test_sign_deploy_account_transaction(gateway_account):
     assert signed_tx.class_hash == class_hash
     assert signed_tx.contract_address_salt == salt
     assert signed_tx.constructor_calldata == calldata
+
+
+@pytest.mark.asyncio
+async def test_deploy_account(
+    client, deploy_account_details_factory, base_account_deploy_map_contract
+):
+    address, key_pair, salt, class_hash = await deploy_account_details_factory.get()
+
+    deploy_result = await Account.deploy_account(
+        address=address,
+        class_hash=class_hash,
+        salt=salt,
+        key_pair=key_pair,
+        client=client,
+        chain=StarknetChainId.TESTNET,
+        max_fee=MAX_FEE,
+    )
+    await deploy_result.wait_for_acceptance()
+
+    account = deploy_result.account
+
+    assert isinstance(account, BaseAccount)
+    assert account.address == address
+
+    # Test making a tx
+    res = await account.execute(
+        calls=Call(
+            to_addr=base_account_deploy_map_contract.address,
+            selector=get_selector_from_name("put"),
+            calldata=[30, 40],
+        ),
+        max_fee=MAX_FEE,
+    )
+    _, status = await account.client.wait_for_tx(
+        res.transaction_hash, wait_for_accept=True
+    )
+
+    assert status in (
+        TransactionStatus.ACCEPTED_ON_L1,
+        TransactionStatus.ACCEPTED_ON_L2,
+    )
```

Comment:
Do we have more code like this in the code? I think we could use a custom `assert_tx_passed(tx_hash)` for something like this.

Reply:
`wait_for_accept` especially when you add a `wait_for_accept=True` parameter basically asserts that so in most places we don't check the status of transaction (tests would fail if it were rejected). I added status assert here just to be extra sure.